mujoco_warp/_src/smooth.py

```diff
@@ -105,15 +105,19 @@ def _kinematics_level(
 
     # mocap bodies have world body as parent
     mocapid = body_mocapid[bodyid]
-    if pid == 0 and mocapid != -1:
+    if mocapid >= 0:
```

Comment:
perhaps with this, we can now afford a simplifcation and maybe even save a few registers:

```py
mocapid = body_mocapid[bodyid]
if mocapid >= 0:
  xpos = mocap_pos_in[worldid, mocapid]
  xquat = mocap_quat_in[worldid, mocapid]
else:
  xpos = body_pos[body_pos_id, bodyid]
  xquat = body_quat[body_quat_id, bodyid]

if pid >= 0:
  xpos = xmat_in[worldid, pid] @ xpos + xpos_in[worldid, pid]
  xquat = math.mul_quat(xquat_in[worldid, pid], xquat)
```

WDYT?